nixos/modules/services/databases/redis.nix

```diff
@@ -186,9 +186,9 @@ in
 
   config = mkIf config.services.redis.enable {
 
-    boot.kernel.sysctl = mkIf cfg.vmOverCommit {
-      "vm.overcommit_memory" = "1";
-    };
+    boot.kernel.sysctl = {
+      "vm.nr_hugepages" = "0";
+    } // mkIf cfg.vmOverCommit { "vm.overcommit_memory" = "1"; };
```

Comment:
Merging modules with `//` does not work and will need to be fixed using `mkMerge` like in https://github.com/NixOS/nixpkgs/pull/72916.

Reply:
Ah, I didn't know that. What is the explanation?

Reply:
Nix module system uses `_type` attributes to tag attribute sets:

```nix
nix-repl> lib.mkIf false { foo = 1; }
{ _type = "if"; condition = false; content = { ... }; }
```

So if you merge it into a different attribute:

```nix
nix-repl> { bar = 5; } // lib.mkIf false { foo = 1; }
{ _type = "if"; bar = 5; condition = false; content = { ... }; }
```

the bar will not be properly merged into `content` attribute.

We really need [nixos-pills](https://github.com/NixOS/nix-pills/issues/120).

Reply:
followed up in https://github.com/NixOS/nixpkgs/pull/73049.